components/net/http_loader.rs

```diff
@@ -844,8 +844,14 @@ pub fn http_fetch(
                     .ok()
             });
 
-        // Substep 4.
-        response.actual_response_mut().location_url = location;
+        // Substep 4.               
+        response.actual_response_mut().location_url = location.map(|res| res.map(|mut url| {
```

Comment:
I think this would be more clear as:
```rust
let mut location = ...;

// Substep 4.
if let Some(ref mut location) = location {
    if location.fragment().is_none() {
        let current_url = request.current_url();
        location.set_fragment(current_url.fragment());
    }
}
response.actual_response_mut().location_url = location;
```

Reply:
The inner most `location` is a `Result` which must be destructured (more nesting) or mapped over (may be less readable). Mutating in place is probably preferable to cloning however:

```rust
        // Substep 4.               
        if let Some(ref mut location) = location {
            if let Ok(ref mut location) = location {
                if location.fragment().is_none() {
                    let current_url = request.current_url();
                    location.set_fragment(current_url.fragment());
                }
            }            
        }
        response.actual_response_mut().location_url = location;
```